### PR TITLE
Android build for master branch

### PR DESCRIPTION
**Pull request by mshabunin**
- ported all manager changes from 2.4
- fixed manager build (visibility issues)
- added 3.0.0 to supported library list (and 2.4.10 - 2.4.11)
- android examples try to load OpenCV 3.0.0

__Note:__ mechanism for loading both library versions (2.4 and 3.0) is not implemented yet
__Note:__ built with NDK r8e, gcc-4.4.3 toolchains

### CODE REVIEW

**Comment by PhilLab** [comment]
nice! I was waiting for that, thanks :smiley:

**Comment by vpisarev** [comment]
@asmorkalov, could you please review it?

File: platforms/android/service/engine/AndroidManifest.xml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.opencv.engine"
-    android:versionCode="218@ANDROID_PLATFORM_VERSION_CODE@"
-    android:versionName="2.18" >
+    android:versionCode="219@ANDROID_PLATFORM_VERSION_CODE@"
+    android:versionName="2.20" >
```
Comment (asmorkalov) [inline platforms/android/service/engine/AndroidManifest.xml]: Let's use 3.x and 300@ANDROID_PLATFORM_VERSION_CODE@ versions to differentiate it with 2.4.

**Comment by vpisarev** [comment]
let's put it in and, if needed, fix further with subsequent PRs. :+1:

**Pull request merged**